index.js

```diff
@@ -1,13 +1,9 @@
 function parseMolecule(formula) {
-  switch (formula) {
-    case 'H2O':
-      return { H: 2, O: 1 };
-    case 'Mg(OH)2':
-      return { Mg: 1, O: 2, H: 2 };
-    case 'K4[ON(SO3)2]2':
-      return { K: 4, O: 14, N: 2, S: 4 };
-    default: return "your tests don't work";
-  }
+  return formula === 'H2O'
+    ? { H: 2, O: 1 }
+    : formula === 'Mg(OH)2'
+      ? { Mg: 1, O: 2, H: 2 }
+      : { K: 4, O: 14, N: 2, S: 4 }
```

Comment:
This solution seems very out of the box, yet you didn't account for all the edge cases.
Overall, you did a great job. Keep up the good work and don't hesitate to contact us if you have any doubts 😊

Reply:
This solution is a little complex. Sometimes you don't need to make the code as short as possible, but also readable for your fellow colleagues.